src/base/bittorrent/session.cpp

```diff
@@ -2387,6 +2386,14 @@ void Session::setBannedIPs(const QStringList &newList)
                         , Log::WARNING);
         }
     }
+    // now we have to sort IPs and make them unique
+    filteredList.sort();
+    filteredList.removeDuplicates();
```

Comment:
These 2 should be moved at the top before the check:
```c++
if (newList == m_bannedIPs)
return; // do nothing
```

Also remove the commented out lines.